docs/guides/result_storage.mdx

```diff
@@ -0,0 +1,95 @@
+---
+id: result-storage
+title: Result storage
+description: How to store and retrieve data with KeyValueStore and Dataset in Crawlee
+---
+
+import ApiLink from '@site/src/components/ApiLink';
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import CodeBlock from '@theme/CodeBlock';
+
+import DatasetBasicExample from '!!raw-loader!./code/result_storage_dataset_basic.py';
+
+import KvStoreBasicExample from '!!raw-loader!./code/result_storage_kvstore_basic.py';
+
+import RsPurgeExplicitlyExample from '!!raw-loader!./code/result_storage_purge_explicitly.py';
+
+This guide explains the available result storage options in Crawlee and how to use them to store and retrieve data.
+
+## Result storage overview
+
+Crawlee offers various types of result storage, each tailored for specific tasks. By default, data is saved to a local disk in the directory specified by the `CRAWLEE_STORAGE_DIR` environment variable. If this variable is not set, Crawlee will automatically use `./storage` within the current working directory.
+
+The `MemoryStorage` class manages Crawlee's storage. Throughout the crawler's execution, all data is held in memory, and it is simultaneously written to local files within their respective storage type folders.
+
+## KeyValueStore
+
+The key-value store serves the purpose of saving and retrieving data records or files. Each data record is identified by a unique key and linked to a specific MIME content type. Key-value stores are particularly suited for storing web page screenshots, PDFs, or for maintaining the state of crawlers.
+
+Every Crawlee project run is tied to a default key-value store. By convention, the project’s input and output are saved in this default key-value store under the keys `INPUT` and `OUTPUT`, respectively. Typically, both input and output are in JSON format, though other formats are also acceptable.
+
+In Crawlee, the key-value store is represented by the KeyValueStore class. To facilitate easy access to the default key-value store, Crawlee provides the functions `KeyValueStore.getValue()` and `KeyValueStore.setValue()`.
+
+The data is stored in the directory specified by the `CRAWLEE_STORAGE_DIR` environment variable as follows:
+```text
+{CRAWLEE_STORAGE_DIR}/key_value_stores/{STORE_ID}/{KEY}.{EXT}
+```
+- `{CRAWLEE_STORAGE_DIR}`: The root directory for all storage data specified by the environment variable.
+- `{STORE_ID}`: The identifier for the key-value store, either "default" or as specified by `CRAWLEE_DEFAULT_KEY_VALUE_STORE_ID`.
+- `{KEY}`: The unique key for the record.
+- `{EXT}`: The file extension corresponding to the MIME type of the content.
+
+The following code demonstrates basic operations of key-value stores:
+
+<Tabs groupId="keyvaluestore">
+    <TabItem value="kvstore_basic_example" label="Basic usage" default>
+        <CodeBlock className="language-python">
+            {KvStoreBasicExample}
+        </CodeBlock>
+    </TabItem>
+</Tabs>
```

Comment:
Why tabs if you have there only one item? That does not make sense. See how this is done in the [Request storage guide](https://crawlee.dev/python/docs/guides/request-storage).

Reply:
I mean, could you please add other tabs as we have for the [Request storage guide](https://crawlee.dev/python/docs/guides/request-storage)? Users should find out here how to use the result storages with crawlers.

- Basic usage
- Usage with Crawler
- Explicit usage with Crawler